### PR TITLE
docs: Add releasing document

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+$ git push origin v0.x.0
+```
+
+6. Push tag for real release
+
+```bash
+$ git push origin v0.x.0
+```
+
+7. Once the CI is done, go to https://github.com/inspektor-gadget/inspektor-gadget/releases and edit the created release adding the notes you prepated before
```
Comment (alban) [inline docs/releasing.md:75]:
```suggestion
7. Once the CI is done, go to https://github.com/inspektor-gadget/inspektor-gadget/releases and edit the created release adding the notes you prepared before
```

File: docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+2. Tag a release canditate version
+
+```bash
+$ git tag v0.x.0-rc
+```
+
+3. Push tag to remote
+
+```bash
+$ git push origin v0.x.0-rc
```
Comment (mauriciovasquezbernal) [inline docs/releasing.md:61]: @eiffel-fl I noticed it would create a release on Github. I had to remove this by hand later on. Does anybody have an idea how to fix it?

Comment (alban) [reply docs/releasing.md:61]: You could add the following in the relevant Github action. We already do something similar for the krew-index action.
```diff
-if: startsWith(github.ref, 'refs/tags/v')
+if: startsWith(github.ref, 'refs/tags/v') && !endsWith(github.ref_name, 'rc')
```

Comment (mauriciovasquezbernal) [reply docs/releasing.md:61]: That can work, but then what's the point of creating a release candidate?

IMO we can avoid this step and test changes to the release job in a fork. This to avoid notifying users about a rc candidate version. @eiffel-fl what do you think?

Comment (mauriciovasquezbernal) [reply docs/releasing.md:37]: > The whole point for me is to test and ensure everything is OK.

Almost all the testing is done in main when a PR is merged. If the tests are passing in main, the only thing that could break is the release job. In that case I'd suggest to test changes modifying it in a fork before merging them.  

> Will you be able to test on AKS by using a fork? 

That's right, one should configure the secrets in the repo to do it.

> I do not think this a bad thing to have release candidate available for everyone.

I agree that release candidates are good, however I think they bring little value as we're using them in our project: to check if the release pipeline is working fine.

A release candidate is a version that you release, let some users test it and then you fix the bugs they find, however in our case we're just release the final version after the rc. 


In summary, I feel creating a release candidate adds extra complication and noise to the release and doesn't bring very big benefits, hence I think we should remove this. What do you think?

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+weight: 20
+description: >
+  How to release Inspektor Gadget
+---
+
+The Inspektor Gadget release process is heavily automated, however some manual steps are still
+required. This guide describes the process to make a new release of Inspektor Gadget and includes
+some information of what to do in case something goes wrong.
+
+Be sure you have a general understading of the whole release process before starting to follow the
```
Comment (alban) [inline docs/releasing.md:12]:
```suggestion
Be sure you have a general understanding of the whole release process before starting to follow the
```

File: docs/releasing.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+steps described here. It's also important to stay aware as a mistake could cause issues for our
+users.
+
+## Making a new release
+
+0. Be sure tests on main branch are passing, if they are not passing it's needed to fix them before.
```
Comment (alban) [inline docs/releasing.md:18]:
```suggestion
0. Be sure tests on main branch are passing, please fix them if not.
```

File: docs/releasing.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+### Bug Fixes
+### Documentation Improvements
+### Testing and Continue Integration
+```
+
+8. Verify that the CI created a pull request in
+   [krew-index](https://github.com/kubernetes-sigs/krew-index/pulls) and that it was merged
+   automatically be the bot
+
+9. Send an announcement on the `inspektor-gadget` Slack channel
```
Comment (alban) [inline docs/releasing.md:71]:
```suggestion
9. Send an announcement on the [#inspektor-gadget](https://kubernetes.slack.com/archives/CSYL75LF6) Slack channel
```

File: docs/releasing.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+### The CI process for the tag failed
+
+It can happen that the CI fails because of a flaky test, it's fine to rerun the failed jobs to make
+them pass.
+
+### The krew-index PR wasn't merged automatically
+
+In some situations the PR is not merged by the bot, in that case it's necessary to ping the
+maintainers to manually merge it.
```
Comment (alban) [inline docs/releasing.md:83]:
```suggestion
Krew maintainers to manually merge it.
```

File: docs/releasing.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
+---
+title: Releasing
+weight: 20
```
Comment (alban) [inline docs/releasing.md:3]: Could this page be moved in a "Development" section so it is not mixed with other user documentation in https://inspektor-gadget.io/docs/latest/contributing/?

**Review by alban — COMMENTED** [review]
A few comments. I think this can be merged after that.

Is release-create-tag.png still used somewhere?

File: docs/releasing.md (hunk 7 above)
Comment (mauriciovasquezbernal) [reply docs/releasing.md]: I moved it to the contributing document. Is it what you intended?

Comment (mauriciovasquezbernal) [reply docs/releasing.md]: Makes sense. I updated the PR.

**Review by alban — APPROVED** [review]
Another path to update:
```
$ git grep CONTRIBUTING.md
Dockerfiles/gadget-default.Dockerfile:8:# See BCC section in docs/CONTRIBUTING.md for further details.
```

LGTM after that.
